tests/test_site_apportionment/test_site_apportionment.py

```diff
@@ -58,3 +67,325 @@ def test_create_notnull_mask(self):
+    [4990000252, 202212, "UV1 2WX", 7084, 7058606],
+    [4990000294, 202212, "YZ3 4AB", 3188, 1828472],
+    [4990000336, 202212, "CD5 6EF", 62, 3262548],
+    [4990000378, 202212, "GH7 8IJ", 1180, 9348647]
+    ]
+
+
+    return pd.DataFrame(data=sample_data, columns=sample_cols)
+
+
+def test_count_unique_postcodes_in_col_positive():
+    df = create_category_test_df()
+
+    # Apply the function
+    result_df = count_unique_postcodes_in_col(df)
+
+    # Expected output
+    expected_cols = ['reference', 'period', '601', 'headcount_dummy', '211_dummy', '601_count']
+
+
+    expected_data = [
+        [4990000000, 202212, 'AB1 2CD', 5663, 8855030, 2],
+        [4990000000, 202212, 'B27 2CD', 242, 5949501, 2],
+        [4990000084, 202212, 'EF3 4GH', 8020, 5085659, 1],
+        [4990000126, 202212, 'IJ5 6KL', 3877, 5808144, 3],
+        [4990000126, 202212, 'MN7 8OP', 2756, 8889091, 3],
+        [4990000126, 202212, 'QR9 10ST', 8832, 4722201, 3],
+        [4990000252, 202212, 'UV1 2WX', 7084, 7058606, 1],
+        [4990000294, 202212, 'YZ3 4AB', 3188, 1828472, 1],
+        [4990000336, 202212, 'CD5 6EF', 62, 3262548, 1],
+        [4990000378, 202212, 'GH7 8IJ', 1180, 9348647, 1]]
+
+    expected_df = pd.DataFrame(data=expected_data, columns=expected_cols)
+
+    # Check if the output is as expected
+    pd.testing.assert_frame_equal(result_df, expected_df, check_dtype=False)
+
+
+def create_sample_repeated_postcodes_df():
+    """
+    Creates a sample DataFrame using predefined data and columns.
+
+    Returns:
+        pd.DataFrame: The created DataFrame.
+    """
+    sample_cols = ["reference", "period", "601", "headcount_dummy", "211_dummy"]
+
+    sample_data = [
+    [4990000000, 202212, "B27 2CD", 5663, 8855030],
+    [4990000000, 202212, "B27 2CD", 5663, 8855030], #repeated postcode
+    [4990000084, 202212, "B27 2CD", 8020, 5085659], # repeated postcode but for different ref
+    [4990000252, 202212, "UV1 2WX", 8020, 5085659],
+
+    ]
+
+
+    return pd.DataFrame(data=sample_data, columns=sample_cols)
+
+def test_count_unique_postcodes_in_col_repetitive():
+    """
+    Test the function count_unique_postcodes_in_col."""
+    df = create_sample_repeated_postcodes_df()
+
+    # Apply the function
+    result_df = count_unique_postcodes_in_col(df)
+
+    # Expected output
+    expected_cols = ['reference', 'period', '601', "headcount_dummy", "211_dummy", '601_count']
+
+    expected_data = [
+        [4990000000, 202212, "B27 2CD", 5663, 8855030, 1],
+        [4990000000, 202212, "B27 2CD", 5663, 8855030, 1], # repeated postcode but for the same ref
+        [4990000002, 202212, "B27 2CD", 8020, 5085659, 2], # postcode again repeated but for different ref
+        [4990000002, 202212, 'UV1 2WX', 8020, 5085659, 2] # postcode_count should be 2
+    ]
+
+    expected_df = pd.DataFrame(data=expected_data, columns=expected_cols)
+
+    # Check if the output is not as expected
+    with pytest.raises(AssertionError):
```

Comment:
assert_frame_equal not working here, but not sure why.  If I replace lines 168/169 with assert_frame_equal(result_df, expected_df) it works correctly